WordPress/gutenberg
# Testing: Fail CI build if local changes exist

This pull request seeks to add a failure case to our existing `ci` script to trigger a build failure if local changes exist after one or both of `npm run install` (e.g. unsynced `package-lock.json` changes) or `npm run docs:build`. This has been increasingly problematic as of late due to failures to update generated files upon changes, causing confusion for later developers who are then forced to consider how to reconcile the change.

**Testing instructions:**

Verify that no error is produced upon running `npm run ci` (or, more specifically, `npm run check-local-changes`) after checking out the branch.

Verify that an error code is produced when making a local change (but not committing) and re-running the script.
